cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRuntimeProvider.java

```diff
@@ -68,9 +69,11 @@ public ProgramRunner createProgramRunner(ProgramType type, Mode mode, Injector i
    */
   private ProgramRunner createSparkProgramRunner(Injector injector, String programRunnerClassName) {
     try {
-      CConfiguration cConf = injector.getInstance(CConfiguration.class);
-      final SparkRunnerClassLoader classLoader = new SparkRunnerClassLoader(getClassLoaderURLs(cConf),
-                                                                            getClass().getClassLoader());
+      boolean rewriteYarnClient = injector.getInstance(CConfiguration.class)
```

Comment:
By default, it is always enabled. Without YARN-4727 being fixed, on CM cluster, we still need this even the Spark bug is patched.